src/Workspaces/Core/Portable/Workspace/Workspace_Editor.cs

```diff
@@ -395,26 +395,39 @@ private ISet<DocumentId> GetProjectOpenDocuments_NoLock(ProjectId project)
             return openDocs;
         }
 
-        internal Task OnDocumentOpenedAsync(
+        protected internal void OnDocumentOpened(
             DocumentId documentId, SourceTextContainer textContainer,
             bool isCurrentContext = true)
         {
-            OnDocumentOpened(documentId, textContainer, isCurrentContext);
-            return SpecializedTasks.EmptyTask;
+            OnDocumentOpenedAsync(documentId, textContainer, isCurrentContext, wait: true).Wait(CancellationToken.None);
         }
 
-        protected internal void OnDocumentOpened(
+        internal Task OnDocumentOpenedAsync(
```

Comment:
Yes.  The goal is to move everyone to the async codepath.  I can make that change now as well if you want.